pgdb/api.py

```diff
@@ -232,6 +232,52 @@ def words(self, request, pk=None):
             results = [dict(x) for x in q.aggregate(Count())[:int(count)]]
         return Response(json.dumps(results))
 
+    @detail_route(methods=['get'])
+    def default_subsets(self, request, pk=None):
+        subset_class = request.GET.get('subset_class', 'syllabics')
+        print(subset_class)
+        if request.auth is None:
+            return Response(status=status.HTTP_401_UNAUTHORIZED)
+        corpus = models.Corpus.objects.get(pk=pk)
+        if not request.user.is_superuser:
+            permissions = corpus.user_permissions.filter(user=request.user).all()
+            if not len(permissions):
+                return Response(status=status.HTTP_401_UNAUTHORIZED)
+
+        if subset_class not in ['syllabics', 'sibilants', 'stressed_vowels']:
+            return Response(
+                'Invalid subset class',
+                status=status.HTTP_400_BAD_REQUEST)
+
+        if subset_class == 'syllabics':
+            #TODO: check if corpus.name is prefixed with SPADE(sometimes it seems to be, e.g., for ICE-CAN).

```

Comment:
I would do something like:

```
if corpus.name.endswith('SCOTS'):
```

since that would capture the `spade-*` case, these are built around the original SPADE repo, where the prefix wasn't present.

Reply:
Another way would be to check if it starts with spade and then do

```
corpus_name = corpus.name.split('-', 1)[1]
```

otherwise `corpus_name = corpus.name`

This is probably a better solution.